src/components/LinkableText.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import {Text, Linking} from 'react-native';
+import {COLORS} from '../utils/theme';
+
+interface Props {
+  text: string;
+  style?: object;
+}
+
+export const LinkableText: React.FC<Props> = ({text, ...props}) => {
+  const handlePress = (url: string) => {
+    Linking.openURL(url);
+  };
+
+  // Regular expression to match URLs
+  const urlRegex = /(https?:\/\/[^\s]+)/g;
+
+  // Split the text by URLs
+  const parts = text.split(urlRegex);
+
+  return (
+    <Text style={props.style}>
+      {parts.map((part, index) =>
+        urlRegex.test(part) ? (
+          <Text
+            key={index}
+            style={{color: COLORS.blueLink, textDecorationLine: 'underline'}}
+            onPress={() => handlePress(part)}>
+            {part}
+          </Text>
+        ) : (
+          <Text key={index}>{part}</Text>
+        )
+      )}
+    </Text>
+  );
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `LinkableText` component seems to be well implemented. It uses a regular expression to identify URLs in the provided text and splits the text into parts based on these URLs. Each part is then checked if it's a URL or not, and if it is, it's rendered as a clickable link with a specific style. If it's not a URL, it's simply rendered as plain text.

However, there's a potential performance issue here. The `urlRegex.test(part)` function is called twice for each part of the text - once in the conditional check and once when rendering the part. This could be optimized by storing the result of the test in a variable and reusing it.

Here's how you can improve this:

```diff
      {parts.map((part, index) => {
        const isUrl = urlRegex.test(part);
        return isUrl ? (
          <Text
            key={index}
            style={{color: COLORS.blueLink, textDecorationLine: 'underline'}}
            onPress={() => handlePress(part)}>
            {part}
          </Text>
        ) : (
          <Text key={index}>{part}</Text>
        )
      })}
```

Also, consider handling possible errors that might occur when opening the URL. For instance, if the URL scheme is not supported, `Linking.openURL(url)` will throw an error. You should catch this error and handle it appropriately, perhaps by showing a message to the user.

```diff
  const handlePress = (url: string) => {
    Linking.openURL(url).catch(err => console.error("Failed to open URL: ", err));
  };
```

Lastly, using the array index as a key in React is generally not recommended as it can lead to unexpected behavior in certain cases. If the items in your list can reorder, this can cause issues with component state. If you have a unique identifier for each part, use that instead. If not, and the parts will not reorder, then using the index should be fine.

Overall, the code is well written and follows best practices. The changes suggested above are minor optimizations that could improve performance and error handling.


<!-- This is an auto-generated comment by OSS CodeRabbit -->